apartment.rb

```diff
@@ -1,2 +1,30 @@
 class Apartment
+  attr_accessor :number,  :rooms, :monthly_rent  
+  def initialize(args = {})
+    @number = args.fetch(:number, 0)
+    @rooms = args.fetch(:rooms, 0)
+    @monthly_rent = args.fetch(:monthly_rent, 1000)
+  end
+
+  def total_sqft
+    rooms.map! do |room| 
```

Comment:
You can use the reduce method here to solve this without having to iterate over the array using map and then calling reduce on it.
```ruby
rooms.reduce(0) { |sum, room| sum + room.sqft }
```
The 0 is the starting value. Sum is what is incremented each time and returned at the end. Room is each room in the rooms array. So each iteration, you are adding the room's sqft to the sum value and at the end, returning the sum.